src/main/java/in/casekartin/dao/CaseManagerDAO.java

```diff
@@ -1,87 +1,117 @@
+	public static boolean addCase(String caseName, Float price) throws SQLException,DBException {
+		Connection connection = null;
+		PreparedStatement pst = null;
+		try {
+			// Get Connection
+			connection = ConnectionUtil.getConnection();
+			// prepare data
+			String sql = "insert into caseTypes(casename,price) values (?,?)";
+			pst = connection.prepareStatement(sql);
+			pst.setString(1, caseName);
+			pst.setFloat(2, price);
 
+			// Execute Query
+			int rows = pst.executeUpdate();
+			boolean inserted = false;
+			if(rows==1)
+			{
+				inserted=true;
+			}
+			return inserted;
+		} catch (Exception e) {
+			throw new SQLException("Unable to add case");
+		} finally {
+			ConnectionUtil.close(connection, pst);
+		}
 	}
 	/**
-	 * method for add the case name
-	 * 
-	 * @param caseName
-	 * @param cost
+	 * Retrieve all data from database table 
 	 * @return
+	 * @throws Exception
 	 */
-	public static boolean addCase(String caseName,String cost)
-	{
-		boolean isAdded=false;
-		Float price = Float.parseFloat(cost);
-		
-		if (CaseManagerUtil.isValidCaseName(caseName) && CaseManagerUtil.isValidCost(price)
-				&& CaseManagerUtil.isCharAllowed(caseName) && CaseManagerValidator.isCaseNameNotExist(caseName)) {
-			CaseManager newCase = new CaseManager(caseName.toUpperCase(), price);
-			caseTypes.add(newCase);
-			isAdded=true;
+	public static Set<CaseManager> listAllCases() throws DBException,SQLException {
+		Set<CaseManager> caseTypes = new HashSet<>();
+		Connection connection = null;
+		PreparedStatement pst = null;
+		ResultSet rs=null;
+		try {
+			connection = ConnectionUtil.getConnection();
+			// Retrieve data from table
+			String sql = "select casename,price from caseTypes";
+			pst = connection.prepareStatement(sql);
+			rs = pst.executeQuery();
+			while (rs.next()) {
+				String caseName = rs.getString("casename");
+				Float price = rs.getFloat("price");
+
+				// Store the data in model
+				CaseManager product = new CaseManager(caseName, price);
+				// Store all products in list
+				caseTypes.add(product);
+			}
+		} catch (Exception e) {
+			throw new SQLException("Unable to display case");
+		}
+		finally {
+		ConnectionUtil.close(connection, pst, rs);
 		}
-		return isAdded;
+		return caseTypes;
 	}
 	/**
-	 * method for delete the case name
-	 * 
+	 * Delete specific data in database
 	 * @param caseName
-	 * @param cost
 	 * @return
+	 * @throws Exception
 	 */
-	public static boolean deleteCase(String caseName) {
-		boolean isDeleted=false;
-		CaseManager searchCase=CaseManagerValidator.isCaseNameExist(caseName); 
-		if(searchCase!=null) {
-			caseTypes.remove(searchCase);
-			isDeleted = true;
+	public static boolean deleteCase(String caseName) throws DBException,SQLException {
+		Connection connection = null;
+		PreparedStatement pst = null;
+		try {
+			// Get Connection
+			connection = ConnectionUtil.getConnection();
+			// prepare data
+			String sql = "delete from caseTypes where casename=?";
```

Comment:
Instead of deleting a casetype can you make the status active/inactive like that
It may be needed for future reference